src/org/jitsi/impl/neomedia/RTPConnectorInputStream.java

```diff
@@ -743,9 +743,9 @@ private void runInReceiveThread()
 
         while (!closed)
         {
-            // http://code.google.com/p/android/issues/detail?id=24765
-            if (OSUtils.IS_ANDROID)
-                p.setLength(PACKET_RECEIVE_BUFFER_LENGTH);
+            // Reset the buffer, because the previous call to receive() might
+            // have bumped the offset or even changed the byte[].
+            p.setData(buffer, 0, buffer.length);
```

Comment:
ok, i think just more clarifying questions for my own sake:
1) the octo logic won't always kick in, will it?  does it make sense to have the reset (in the case of stripping the octo buffers) in the octo receive logic so its transparent elsewhere?
2) if i'm understanding what you said correctly (i don't think i am), how is the octo receive bumping the offset ahead different than the normal processing of a packet which reads the buffer?